### ISSUE TITLE
Sometimes, interactables don't trigger their "Untouched" event 

### ISSUE THREAD

**Issue by arnaudcoj** [ExtendRealityLtd/VRTK#2063]
### Environment

* Source of VRTK : GitHub
* Version of VRTK : commit 29d098ce511e9c18a189327b74029b193d6011a9
* Version of the Unity software : Unity 2019.2.5f1
* Hardware used : Oculus Go (but the error is platform agnostic, it works with the simulator too)
* SDK used : Oculus Utilities (same note as above)

### Steps to reproduce

* In the example scene :
* Add an Interactable.Primary_Grab.Secondary_None prefab in the scene
* Place it in front of an object with a collider
* Disable interactable's "use gravity"

* Add a straight pointer from the vrtk prefabs
* Pointer's Follow source = RightcontrollerAlias
* Activation Action = whatever (ie. **SimulatedCameraRig/InputMappings/ButtonControl/Button1

* Add an interactor from the vrtk prefabs in ObjectPointer.Straight/ObjectPointer.Internal/Elements/Destination/Elements.Sphere/ValidContainer
* Change interactor's layer to "Ignore Raycast"

* Add a random gameobject like a cube where you can see it, we'll call it "testcube"
* Disable the object
* In Interactable.Primary_Grab.Secondary_None, add testcube.SetActive(true) in Touched event and testcube.SetActive(false) in Untouched event

* See that when you enable/disable the pointer, the testcube appears/disappears accordingly
* But when the pointer leaves the interactable cube and then touches another collider (ie. the ground), the testcube doesn't disappears. It seems like the Untouched event is not called.
* (NB : it works correctly when we leave the interactable without touching another collider)

### Expected behavior

The interactable should trigger the Untouched event when the interactor leaves it, whatever the new target is.

### Current behavior

When an interactor stops touching an interactable and immediatly touches another object. The first interactable doesn't trigger the Untouched event

**Comment by arnaudcoj** [comment]
Note :

I followed the github template, but here's what I posted in Slack first. There are some gifs, which may be useful to easily understand the issue :

Hi everyone ! I'm trying to make a headset pointer using VRTK4 and I encountered what either is a bug, or me not understanding something, you tell me. :tire_la_langue:
I've got a pointer under the TrackedAlias/HeadsetAlias gameobject, containing an interactor at the end of the raycast.
I've got 2 cubes. A "down cube" which is an interactable object, and a "top cube" which doesn't trigger anything. The two cubes touch each other.
The down cube triggers the activation/disactivation of a third cube (let's call it the "ok cube").
(It may sound complicated but you should understand with the following gifs :tire_la_langue: )
Ok so when the pointer is on the down cube, then leaves the cube or when I disable the pointer it's ok
https://gyazo.com/bfd2c5ac83b82a931961a7a7d628c2ff
But when the pointer leaves the down cube AND points to the top cube, it seems that the Interactable's Untouched event is not called, causing the "ok cube" to stay activated (then when I touch+untouch the down cube it works again)
https://gyazo.com/7f1da7b4fba66ee4b0b347e10472566d
The down cube should trigger the untouched event when the pointer leaves it, whatever the new target is. At least that's what I want here.
I don't know what to do, if you have any advice or if I'm going the wrong way here don't hesitate to tell me !
Thank you ! :visage_légèrement_souriant:

**Comment by thestonefox** [comment]
![pointer](https://user-images.githubusercontent.com/1029673/66389790-262db380-e9c1-11e9-87c1-72fcf03208e4.gif)

it's quite random, not easily recreatable so tracking down the issue is going to be a pain.

**Comment by arnaudcoj** [comment]
I noticed that when I duplicate my first interactable and put the 2 objects next to each others it behaves as in your gif.
But when I create a simple cube from Unity and put it next to the interactable, it does behaves consistently.

Can you confirm this ?

**Comment by thestonefox** [comment]
Yeah I can confirm that with a normal collider it consistently does it after the second collision. That will make it easier to debug.

**Comment by thestonefox** [comment]
I think your initial idea about it being because the interactor is turned off when the pointer is not hitting a valid object is the reason behind why this is happening.

**Comment by thestonefox** [comment]
I know what the problem is

When the Interactor is disabled it doesn't emit the `CollisionTracker.CollisionStopped` event because the rule fails due to the interactor being disabled.

I think the fix will probably include something like passing a parameter to the event emission check to get it to check if the collisionTracker gameobject is active and if it's not then to ignore the rule check. This can then be done only when the collision tracker is disabled.

This basically then says when the collision tracker component is disabled, any collision notifier it is connected to will be told the collision has stopped.

**Comment by thestonefox** [comment]
This issue is actually the Rule is ignored because the rule container is disabled and this means the negation rule returns false because the `RequiresBehaviourState` is false.

Perhaps some way of overriding the rule to ignore state is a better idea?
